tools/configen/tests/test_modules/future_annotations.py

```diff
@@ -26,12 +26,12 @@ class ExampleClass:
     def __init__(
         self,
         no_default: float,
-        lst: List[str],
-        passthrough_list: List[LibraryClass],
-        dataclass_val: List[User],
-        def_value: List[str] = [],
+        lst: list[str],
+        passthrough_list: list[LibraryClass],
+        dataclass_val: list[User],
+        def_value: list[str] = [],
         default_str="Bond, James Bond",
-        none_str: Optional[str] = None,
+        none_str: str | None = None,
```

Comment:
This change is not compatible with python3.7.

Python3.7 doesn't support subscripting the builtin `list`, nor does it support using the bar `|` for union types.

Reply:
It does for typing (but e.g. not type aliases) when using `from __future__ import annotations` as at the beggining of this file.

Reply:
This is because the usage of that annotation makes the types be treated as strings and they are evaluated later.

Reply:
I wonder, can we get away with removing the `from __future__ import annotations` line in this file?

Reply:
It's called `future_annotations`, so sounds like no? 😄

Reply:
Oh, great point.

Reply:
To keep consistency with the rest of the codebase, I still think there's benefit to using `List` instead of `list`. We don't need to convert `List -> list` just because the pyupgrade tool said so.